TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -843,7 +843,7 @@ public override bool VisitNode(IASTVisitor astVisitor)
         }
     }
     // [COBOL 2002]
-    public class TypeDefinition: DataDefinition, Parent<DataDescription>, IDocumentable
+    public class TypeDefinition: DataDefinition, Parent<DataDescription>, IDocumentable, IEquatable<TypeDefinition>
```

Comment:
TypeDefinition and GeneratedDefinition are special cases when it comes to equality mechanism... They should be changed in a new issue. As discussed together, those two should not be modified in #1586